File: src/com/timepath/hl2/VDFDiffTest.java
package com.timepath.hl2;

import com.timepath.steam.io.VDF2;
import java.util.logging.Logger;

/**
 *
 * @author TimePath
 */
public class VDFDiffTest {

    private static final Logger LOG = Logger.getLogger(VDFDiffTest.class.getName());

    public static void main(String... args) {

        /**
         * "VDF" {
         * "Modified" {
         * "Same"	"yes"
         * "Similar"	"one"
         * "Removed"	"yes"
         * }
         * "Removed" {}
         * "Same" {}
         * }
         */
        final VDF2 n1 = new VDF2("A");
        
         /**
         * "VDF" {
         * "Modified" {
         * "Same"	"yes"
         * "Similar"	"two"
         * "Added"	"yes"
         * }
         * "New" {}
         * "Same" {}
         * }
         */
        final VDF2 n2 = new VDF2("B");

        VDF2 tmp = new VDF2("Modified");
        tmp.addAllProperties(new VDF2.VDFProperty("Same", "yes"),
                             new VDF2.VDFProperty("Similar", "one"),
                             new VDF2.VDFProperty("Removed", "yes"));
        n1.addNode(tmp);
        n1.addNode(new VDF2("Removed"));
        n1.addNode(new VDF2("Same"));

        VDF2 tmp2 = new VDF2("Modified");
        tmp2.addAllProperties(new VDF2.VDFProperty("Same", "yes"),
                              new VDF2.VDFProperty("Similar", "two"),
                              new VDF2.VDFProperty("Added", "yes"));
        n2.addNode(tmp2);
        n2.addNode(new VDF2("New"));
        n2.addNode(new VDF2("Same"));

        n1.rdiff2(n2);
        
//        Node.debugDiff(n1.rdiff(n2));
    }

}

File: src/main/kotlin/com/timepath/hl2/VDFDiffTest.kt
package com.timepath.hl2

import com.timepath.steam.io.VDF
import java.awt.BorderLayout
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import java.io.ByteArrayInputStream
import java.io.IOException
import java.nio.charset.StandardCharsets
import java.util.logging.Level
import java.util.logging.Logger
import javax.swing.*
import kotlin.platform.platformStatic

/**
 * @author TimePath
 */
class VDFDiffTest protected constructor() : JFrame() {
    protected var text1: JTextArea
    protected var text2: JTextArea

    init {
        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE)
        setLayout(BorderLayout())
        add(object : JSplitPane() {
            init {
                setResizeWeight(.5)
                text1 = object : JTextArea() {
                    init {
                        setTabSize(4)
                        setText("\"A\" {\n\t\"Modified\" {\n\t\t\"Same\"\t\"yes\"\n\t\t\"Similar\"\t\"one\"\n\t\t\"Removed\"\t\"yes\"\n\t}\n\t\"Removed\" {}\n\t\"Same\" {}\n}\n")
                    }
                }
                setLeftComponent(JScrollPane(text1))
                text2 = object : JTextArea() {
                    init {
                        setTabSize(4)
                        setText("\"B\" {\n\t\"Modified\" {\n\t\t\"Same\"\t\"yes\"\n\t\t\"Similar\"\t\"two\"\n\t\t\"Added\"\t\"yes\"\n\t}\n\t\"New\" {}\n\t\"Same\" {}\n}\n")
                    }
                }
                setRightComponent(JScrollPane(text2))
            }
        })
        add(object : JButton("Diff") {
            init {
                addActionListener(object : ActionListener {
                    override fun actionPerformed(e: ActionEvent) {
                        try {
                            val n1 = VDF.load(ByteArrayInputStream(text1.getText().toByteArray()))
                            val n2 = VDF.load(ByteArrayInputStream(text2.getText().toByteArray()))
                            n1.getNodes()[0].rdiff2(n2.getNodes()[0])
                        } catch (ex: IOException) {
                            LOG.log(Level.SEVERE, null, ex)
                        }

                    }
                })
            }
        }, BorderLayout.SOUTH)
        pack()
        setLocationRelativeTo(null)
    }

    companion object {

        private val LOG = Logger.getLogger(javaClass<VDFDiffTest>().getName())

        public platformStatic fun main(args: Array<String>) {
            SwingUtilities.invokeLater {
                VDFDiffTest().setVisible(true)
            }
        }
    }
}
